### PR TITLE
ViewBar feedback part 1

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

This is almost ready. I would like to add error messages to the JSON editor, and test some things to make sure there are not any major regressions.

Error messages, keyword argument and type annotation placeholders, style and animation tweaks, and some improvements to focusing in and out of parameters and stages.

Enter on an invalid stage or parameter value no longer reverts or deletes stages.

## How is this patch tested? If it is not, in a single sentence please explain why.

Human tested.

### Is this a user-facing change?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

Error messages, keyword arguments and type annotation placeholders, and general form filling improvements have been added to the ViewBar.

### What areas of FiftyOne does this PR affect?

-   [x] `App`: FiftyOne application changes
-   [ ] `Build`: Build and test infrastructure changes
-   [ ] `Core`: Core `fiftyone` Python library changes
-   [ ] `Documentation`: FiftyOne documentation changes
-   [ ] `Other`

### Should this PR be mentioned in the release notes? If so, please choose on category:

-   [ ] `breaking-change` - The PR will be mentioned in the "Breaking Changes"
        section
-   [x] `feature` - A new user-facing feature worth mentioning in the release
        notes
-   [ ] `bug-fix` - A user-facing bug fix worth mentioning in the release notes
-   [ ] `documentation` - A user-facing documentation change worth mentioning
        in the release notes

### CODE REVIEW

**Comment by benjaminpkane** [comment]
This is ready for review. Much of this work will be cleaned up as keyboard shortcuts, and the new focus/commit rules are added.

File: electron/app/components/ViewBar/ViewStage/ErrorMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  useEffect(() => {
+    errorTimeout && clearTimeout(errorTimeout);
+    errorIdTimeout && clearTimeout(errorIdTimeout);
+    errorId &&
+      setErrorIdTimeout(
+        setTimeout(() => {
+          send("CLEAR_ERROR_ID");
+          setErrorTimeout(setTimeout(() => send("CLEAR_ERROR"), 1000));
+        }, 2000)
```
Comment (lethosor) [inline electron/app/components/ViewBar/ViewStage/ErrorMessage.tsx:42]: The validation error messages (e.g. `Invalid value. Expected type "Union[float, NoneType]"`) fade away a bit more quickly than I would like (and users might want to copy them). Could they instead fade out when the user starts typing something again (or shifts to another stage)? Not sure how hard that is.

Comment (benjaminpkane) [reply electron/app/components/ViewBar/ViewStage/ErrorMessage.tsx:42]: I like that. I'll work on it. Thanks!

**Comment by benjaminpkane** [comment]
> One minor issue I noticed: selecting "Take", entering a value, then pressing enter in the seed cell when it's blank gets rid of the value (and then the stage, if it loses focus). Once the stage has been finalized, pressing enter in an empty seed cell works.

Thanks for the review. I'll work on these changes.

**Comment by benjaminpkane** [comment]
I pushed some error message tweaks, per your suggestions. It will be easier to resolve the issue you mentioned with "Take" in my second `ViewBar` PR.

Right now, you have to double click off of a stage error in order to make it disappear. Again, this issue will be easier to resolve in my the next PR.

I hope to have it up today or tomorrow morning, but I can't make any promises.

**Pull request merged**